### DISCUSSION TITLE
Latest Vaillant config files?

### DISCUSSION THREAD

**Discussion by chrisbev** [john30/ebusd#1252] [Q&A]
Hi newbie hope this is OK to post here.
I'm looking for latest config files that includes SetModeOverride as 2.1.x does not have that.  I've searched and found a line to add to the bai.308523.inc file but wondered if someone has this (or more) included or working on this please? Just started, Vaillant 625 ecoTEC plus 001003612 boiler, bit of a learning curve at the moment.

**Comment by JonesGW-MGD** [comment]
2.1.x are the latest 'official' repository files and as you say, don't include the custom setmodeoverride message definition.  To use it therefore, you'll need to gitclone / copy the configuration files to your local instance, where you can edit bai308523 (if that is indeed the boiler file your ebusd instance loads at start) and add the line definition if that makes sense?

**Reply by chrisbev** [reply]
Thanks, yes makes sense, I have the missing line to add.  Are you able to point me to any 'unofficial' files or anyone working on talking to Vaillant boilers?

**Reply by JonesGW-MGD** [reply]
Your best bet may be to check out this forum and the configuration file forum - https://github.com/john30/ebusd-configuration/issues

Form my experience, people have tended to fork the official repository for more 'recent' heating devices such as heat pumps and air recovery units - as you can see from the last updated date of bai.308523.inc, the boiler files tend to be more stable - which typically means that they reflect 'most things' that have been well-tested in terms of not causing issues.  

There maybe a reluctance to add setmodeoverride because of the potential 'harm' it can do.  (On a similar vein, a recent update to the heat pump files which ostensibly was only meant to read values from the display screen appeared to cause havoc with many people's devices....)

What kit do you have and what do you think you are missing in terms of what you want to be able to do?

**Reply by JonesGW-MGD** [reply]
I have a Vaillant IQ 835 which loads the same file and I've managed to get a few 'unknown' messages to report properly by reviewing against other Vaillant boiler files in the repository and trying alternative PBSB values for example - I'm not really missing anything  or yearn for but YMMV of course!

**Reply by chrisbev** [reply]
I've an ecoTEC plus 625 0010036012 (updated ecoTec released end of last year). Only know of SetModeOverride missing at the moment but wanted to start with most up to date files. At moment have loads of unknowns so trying to see how the scan is done to see which reads are tried and how to stop them.  In log it looks at some .csv files and to 08.bai.csv for ID bai00 SW1104 HW7603 then 5 reads all of 0010036012 1300 705937 NB.  Googleing for idiots guide but think will leave for the moment and try SetModeOverride as if that doesn't work then will stop talking to the boiler, get it going then tidy up!  Is PSPB the bai.308523.inc file lines?

**Reply by JonesGW-MGD** [reply]
on start-up, ebusd looks at the (slave) addresses on the bus and seeks to match against configuration files.  Your boiler presents a slave of 08.  Ebusd scans through 08.bai.csv which is like a manifest - you can scan yourself to see that if it finds various product IDs or Hardware versions it loads specific files and if it reaches the end of the file and hasn't found a match, it loads bai.308523.inc - which you can consider is a default Vaillant boiler file.

It also sees whether you have controllers (like a SENSOComfort or SENSOHome) for example) and bus enabled wiring centres like a VR71 etc.  and loads appropriate config files for these.

You can follow the logs to see what gets loaded or you can use ebusctl.  (Depending on how you run ebusd it can be as simple as issuing it from the terminal.  If you're running HA - it's a bit more complicated because you first need to 'jump' onto the container that is running ebusd (say if you have the LukasGrebe add-on)

If former, try:

```
ebusctl info
```

if latter, try

```
 docker exec -it `docker ps | grep ebusd | awk '{print $1}'` /bin/bash
```

and then the above ebusctl command.

If you are using HA, you may find (the first part) my write-up of my learning curve a couple of months ago handy (or not!):

https://github.com/john30/ebusd/discussions/1177

**Reply by JonesGW-MGD** [reply]
The 'PBSB'. is (AFAIU) the 'command sequence'.  For example, in my edited bai.308523.inc,  I changes the 2 hex characters for a number of messages that then provided me with correct data:

```
# d.00
r;wi,,d.00_PartloadHcKW,d.00 heating partload,,,,"6C00",,,power,,,Adjustable partial heat load (Auto: Product automatically adjusts max. partial load to current system demand) -changed from 0704 others to try 9600 & B000
# d.01
r;wi,,d.01_WPPostrunTime,d.01 central heating overruntime,,,,"6400",,,minutes0,,,Overrun time of internal heating pump for heating mode 1 - changed from F703
# d.02
r;wi,,d.02_BlockTimeHcMax,d.02 Max blocking time CH,,,,"2100",,,minutes0,,,Max. burner anti-cycling time heating at 20 °C flow temperature - changed from 0904
```

(Diagnostic code 0 - PartloadHcKW is only partially correct using 6C00- it shows '25' instead of 'Auto' but before it showed 'unknown'.

I didn't decode the messages, simply checked the same lines in other boiler config files for other values.

(my file is now listed by diagnostic code in order - I found that easier to compare against the boiler panel than the way it's laid out in the repository)

**Reply by JonesGW-MGD** [reply]
Although I guess a starting question would be:

What is it you want to use SetModeOverride to do?

**Reply by chrisbev** [reply]
Thanks, a nice insight into diagnostic codes etc.  I'll search for your write-up.  I plan to change the boiler set point for heating leaving it high for dhw (later take in outside temperature).  The system has hive and uses the boiler rt terminal, I have the 3 boiler call signals going to a raspberry pico (dhw, radiators and underfloor heating zones) also added a flow temp sensor to the pico and I tried cutting the rt signal to switch the boiler off in an attempt to control the temperature but not very good as the boiler then times out (anti cycling) so rather a big oscillating cycle of temperature!  I wondered if the  eBus could be used to set the boiler temp or modulation but found most info around uses HA so going that way, may move to pico if can find messages and how to drive the ebus adapter from pico, leaving ha to do the monitoring/graphing?  Later decided to avoid controlling the modulation and control set point thus avoiding the boiler internal actions.  Talking of which for info the 625 boiler replacing a 428 didn't go well can tell more if you are interested?

**Reply by JonesGW-MGD** [reply]
Interesting!  

First, (not that I've done it)  you should be able to use SetModeOverride as you intend I would have thought, with a payload of something like 0;65;-;-;-;0;0;0;0;0;0;0 (for a flow temp of 65 obvs, but it does also rather depend on whose 'flavour' of SetModeOverride you chose to use - one contributor's has 10 parms and the other 12 I think from memory) for example 
https://github.com/john30/ebusd/discussions/518#discussioncomment-8582823

Maybe it's getting a bit off-topic but I guess you plan to monitor the call for heat from your three zones and then issue an appropriate desired flow temp command?  One thing to bear in mind, even if you don't have any external Vaillant controllers, I suspect you'll find the onboard controller may revert to default every 10 mins - that seems to be what people have found (although I have no idea if that is still the case if you have x100 (rt) connected) - if so, you'll need some kind of HA automation or Cron job on the pico, depending on which route you go.

When I had my boiler installed last year (replacing electric storage heaters!) I became something of a reluctant heating expert as I was underwhelmed by the design skills of installers here in the UK.  I ended up calculating room by room heat loss  (to be fair, installers are screwed because the time it takes to do this properly isn't economically viable - certainly not before a quote), designing the pipe runs, (over-) sizing the rads (for as low a flow temp as I could get) and specifying the controls and have been very happy with the result.

One thing I learnt early in my research from Youtube gurus(!) such as https://www.youtube.com/@UrbanPlumbers is only install Vaillant controls with Vaillant boilers!  Consequently, I went overboard and have Vaillant weather comp, a VRC720rf controller and AmbiSENSE TRVs on every rad (13 in total) - hugely expensive - but given I was coming from electric heating.....

I also only have 1 heating zone (as the boiler sees it; 13 of course in reality), and the boiler is a Combi, not quite the complexity you're facing zone-wise.  (Does the UFH blend down physically itself - presumably it does so you can run Rads and UFH together?).  And yes, even more off-topic but interested in your experiences with the 620 - the new range were launched just before I had my Ecotec 835 Exclusive IQ installed, and I was rather miffed at how much nicer the front panel was but stuck with the IQ as I didn't want to annoy my installer even more!

**Reply by chrisbev** [reply]
Yes the UFH has a manual blend valve and running it with my crude temp control at the moment until eBus running.  At the moment on DHW call, all zones get top end boiler temperature I'm sort of deciding/thinking if to stop radiator heating when DHW calling or say give it one minute on and one minute off or a complex different ratio depending on outside temperature may be over the top?  Will decide when it is cold again? UFH can stay running with own blender?  What does your compensation addon do or does the boiler have 2 water outputs?
Decided on a Vaillant approved installer to get the extra warranty with filter (brit gas who failed the existing boiler (we've only been in the house for just over a year!) had much less warranty except if one continually pay for their maintenance plan!
I steered the installer to go for the upgraded ecoTEC plus with larger modulation range of 10:1 as the boiler being replaced 428 with 3:1 range cycled when only 1 hive trv was calling.  So a 6xx rather than a 4xx, I removed the house pump and pressure vessel as 625 has internal ones.  Installer happy with the house system/layout and suggested 625 slightly smaller but we had just had cavity wall insulation.  All seemed good but in use at times the boiler didn't like it after starting after stopping with a different load from that when it was last on and at times the flow temp (i.e. driven by low modulation) stayed low at around 40C!, it was as if it remembered previous load conditions and didn't update/control?   Installer suggested reducing ufh pump speed as could be robbing boiler but didn’t help.  Installer couldn't say if an external bypas should be included or what setting?  Found by lucky circumstances turning the boiler off at mains or a power button reset it then worked OK for a while but after 3 or so days went back to being problematic.  It seemed like a firmware problem and web search didn’t reveal anything and with no suggestions from the installer thought about the IT solution factory reset!   I looked in the settings and there were what looked like gas valve calibration values so didn’t want to try a factory reset but on contacting the installer he said to factory reset was OK as it wouldn't change settings.  Noted all values and factory reset.  Initially OK but after a day or so back to bad behavior and Installer then said to contact Vaillant telephonesupport.  They seemed to appreciate the problem, said an external bypass is not strictly required but no harm as internal one could need some help and should be set just above the internal one.  Interestingly the actual boiler return at the boiler is not the same as the boiler internal sensor as boiler has an internal bypass and conttolled with pump speed via whatever algorithm?  Internal return sensor is at i/p to heat exchanger.   He talked me how to get to the D.xxx codes (although I had already looked there I  just followed what he said) he said to take one setting out of auto and make a fixed value.  D.000 to 25.5kW max load, apparently in auto the boiler makes a decision based on sensors (flow & return temps, modulation, water flow rate, pump speed, I presume) and drives modulation and pump speed.  It then seemed fine driving to set point until....
Next problem after more load changes sometimes it stayed high modulation and not changing quickly enough so went to high temp cutout along with the long anticycle cycle.  Vaillant service visit, seemed a very good guy and confirmed again the boiler should drive the 3 zones, ufh with blender and bypass without problems.   He then did another auto setting change to manual!   This time D.170 hydraulic operating mode to 4 (fixed pump stage) and D.175 pump stage to 80%.  He said boiler may be more noisy and he would probably be back (so not sure of his fix I thought?), but the pump speed didn’t seem to make much difference to noise, noise does change with fan speed which it previously did.
Maybe that is why the wide modulation was only in the 6xx series and hadn't been rolled out to a upgraded 4xx boiler needs more field testing?
Well that is a few words hope they are readable?

**Reply by JonesGW-MGD** [reply]
I feel your pain!  

(Remember, I'm not a heating engineer so take that into account!). I can share some thoughts about what I might do with ebusd in your position and why I think some of the advice you've been given is plain wrong but I'm still not sure what I'd try to fix first - as this veers away from ebusd and more about the behaviour of your specific system, I really recommend you post identical queries about these issues on plumbersforums.net and diynot.com - both have esteemed experts and amateurs alike interested in helping with such issues - and with a lot of Vaillant knowledge too.

First, in terms of using ebusd to implement a control strategy, I would try to mimic PDHW (like 'X Plan' or indeed a Combi).   https://heatingacademynorthampton.co.uk/pdhw/ So, basically a call for DHW shuts off Heat.  It's highly unlikely you'll notice unless it's -10 outside!  That way you'll get the lowest possible return temp, and the highest possible condensing.

(Obviously I don't know your system details, pipe runs, emitter sizing etc. so the flow temp numbers won't necessary be anything more than an example)

For example:

Call for DHW - Set flow temp to 60-65, disable UFH and Rad ZVs
Call for UFH only - Set flow temp to 40-45 (or whatever your UFH heating input should be - no need to waste heat by blending)
Call for Rads only - Set flow temp to 50-55 (assuming you can get away with this - otherwise higher of course)
Call for Rads and UFH - Set flow temp to 60

You could ultimately get smarter - Call for DHW, set flow to 55-60 and then run a anti-legionnaire's cycle at 65 every 24 hours!

(One thing to consider and not ebusd related at all, but if your cylinder doesn't have a restrictor valve (i.e. gate) on the return, might be worth fitting.  It's not common practice, but it generally means that most cylinder setups in the UK don't exchange as much heat as they could as the DHW flow goes through the coil too quickly (unless you have a 'high recovery' coil) - consequently it goes to return at too high a temp to be optimal.   It is possible that a modulating pump might help with this though - a bit outside my knowledge base)

(To help with the monitoring/tweaking of this what you might find it helpful to do is build a couple of flow/return sensors - sounds like you're capable!  I have a couple of sets - ESP32s (running in my case an Arduino sketch) with two DS18b20 probes per device, wifi connected, publishing to HA via MQTT - basically 10 quid each device!). (and yes, I have one of mine on the flow and return near the boiler and reads quite different from D.040 and D.041.)

<img width="2116" alt="image" src="https://github.com/john30/ebusd/assets/68991992/f1969cf8-ed58-401f-95cf-f464c72cd121">


Now some stuff I don't understand - 

You shouldn't need to range rate a system boiler (unless it wasn't sized properly and you really need a 610) with a cylinder (D.000) so no reason why it shouldn't be 'Auto' - range rating is for Combis which may need very high outputs for DHW but less for CH.  E.g: My 35KwH combi 'only' needs a max of 16KwH (2.5 times UK average!) to heat the house to design temp (21 deg at -2 outside) - even so, I have mine set to Auto and seems fine.

But even stranger - how can you range rate a 20Kw boiler for 25.5Kw?  Have I missed something?!

Even if you range-rate it for 15, that's the max CH load - it's still going to use the same algorithms and pump controls to attempt to optimise modulation, flow temps I think?

Regarding the behaviour you observe re. 'modulation memory' - that's potentially above my technical paygrade - the problem is there are so many setting that inter-relate!  Just to check - you have D.017 set to 'flow' I presume?  In 'normal operation' if you monitor D.029 (flow rate l/s) - this is what ebusd/HA is great for btw!  does it seem 'sensible'?  Is D.071 (Max flow rate) and D.128 (min flow rate) still factory defaults?

Also my understanding re modulation differs from your comment: _...  the flow temp (i.e. driven by low modulation) stayed low_. I thought the boiler would fire high, meet set point and then modulate down to maintain set point, not the other way around?

(This is your installation manual in case you don't have it: https://professional.vaillant.co.uk/downloads/aproducts/new-ecotec-plus/firstspirit-16874195715150020308118-05-2681222.pdf - codes at the end)

On pumps - 

In general my basic understanding is you want pumps to work as little and as low as possible while maintaining your design criteria - i.e. keep you warm!

So - on your  TMV/Pump set - my u/s is that the pump is 'sucking' the water around the UFH loops as the boiler pump would be unable to 'push' it around.  (Page 52 of the manual does suggest the boiler has some capability to satisfy UFH without a secondary pump but might be too limited - no idea how much pipe/area you are covering).  Consequently it's after the TMV/blender - I'd still set this as low as possible so that you get the heat transference and a low return temp but I wouldn't have thought it would significantly interfere with the boiler pump - is the manifold very close to the boiler?

Setting D.170 to 'fixed' makes absolutely no sense to me.  You've got a sophisticated boiler with a modern pump - and basically the advice has been to cripple it like a pump from the 1970s?  The only scenario where I could see setting the pump to fixed is if you have hydraulic separation - either a buffer tank or low loss header for example.  In this scenario you're simply interested in the Boiler>LLH>Boiler loop being constant and the off board pump(s) then handle the heating circuits which are hydraulically separated.  (and your manual more or less confirms that - page 50)

And 80% seems high -  my pump is on auto (I know my set up is different) and this is the last 7 days pump % and modulation:

<img width="1051" alt="image" src="https://github.com/john30/ebusd/assets/68991992/b310ae0c-9992-4e9e-b356-89dd7c14578b">

I have a 'lot of water' - 7 out of 13 are high capacity radiators (over-sized Zehdner Charlestons), and miles of MLCP and copper - probably 250-270 litres at least.

If you monitor d.015 is it actually operating at 80% when it's on?

I'd be inclined to check the manual and try setting the pump to default - do you know what it was: 1?
I'd also be inclined to set burner anti-cycle and pump overrun back to defaults too (D.002 & D.001)

And then maybe make small adjustments after having ebusd/HA monitor some of this stuff for a while?

(just to answer your weather comp query - the boiler uses a heat curve to take into account outside temp with desired (room) temps - essentially it determines the DesiredFlowTemp - which is why with weather comp you don't want to be manually changing this).  The practical implication of this is the flow temp is lower when the outside temp is higher and therefore the resulting return temp and condensing benefits are better.  A lower heatcurve is going to mean a lower flow temp and the heating taking longer to reach desired room temp of course.  If I had two or more heating zones, each could have their own heat curves - again I suspect you'd really need to use Vaillant controls to make this work, but as you say you can certainly emulate weather comp by implementing a simplified curve using ebusd - the typical Vaillant curves are on page 52 of your installation guide. I run my heatcurve at about 1.6 (and tweak it if it is going to be arctic to 2) - the Vaillant TechSupport told my installer to set this to between 2.5 and 3 as my old house couldn't cope with it being lower.  I hate to say it but I think the standard of heating and hydraulics in the heating industry in the UK is worse than poor - we've had decades of installers smashing in massively over-sized boilers because 'they work' - The heat loss in the 'average' house in the UK is between 6-7Kw - when have you ever seen a boiler that small?  Modulation of course helps make it even easier for installers without understanding it!).

Hope some of this helps!

**Reply by JonesGW-MGD** [reply]
While I'm thinking....  Your internal expansion vessel is 'only' 10l I think - rule of thumb is 10% so might be worth checking your total capacity esp if you have extensive UFH.  It's probably not causing any of the issues you're experiencing but since you're going to the trouble of optimising your heating system, no harm in checking? (You can work on 16x2mm UFH being 0.12l / m, 15mm copper 0.14 l/m and 22mm 0.31 l/m, plus what's in the boiler, and the rads)

I also suspect your Hives may not helping.  From what I recall, the Hive devices have little or no hysteresis curve - so you set the temp for 20 and boiler calls for heat, the room reaches 20 (or likely overshoots a bit with a fast running pump) but when it falls to 19.8 or 19.5 (I've seen various reports of 0.2-0.5, it kicks the boiler back in.  I think this is less of a problem with a room stat controlled environment but where you have Hive TRVs on Hive's Heating on Demand setup which allows the TRVs to essentially call for heat, then it might be making it more sensitive to cycling - with the temp from the TRV near the metal mass cooling down quicker than the room air near a roomstat and more devices that can effectively call for heat .  If individual TRVs make a call for heat often, and for short periods, the boiler may not get a chance to modulate before it satisfies the desired temp anyway but this is much more conjecture than the earlier response!

**Reply by chrisbev** [reply]
Good thought about expansion vessel will do some sums.  
Nice bit on your system thanks some questions to follow and answers to yours, in the mean time.....
Yes the hive trvs and HoD!  I've plotted out the room temperatures in ha as the hive app only does the thermostat.  My first mod cardboard/foil between the trv and rad helped but a mixture of overshoots or not as the trvs learnt after a few days?  Some good moved to sensitive positions others in don't care position!  Web says the hive trvs have 3 temp sensors, I presume from that one water and 2 air to predict the tempersture gradient and get to in room temperature, a tricky job for the algorithm?  This is what I have so live with it but thinking of adding a room sensor into ha and ramp the trv set point to satisfy the room temperature required?  Yes to me hysteresis very small but annoyingly the HoD works 0.5C below the set point (hive say to stop excessive cycling but to me just moves it to a different point?), imagine evening watching TV all good room temperature set to be comfortable.  Schedule change to get bedroom ready for bed time now bedroom HoD calls for boiler, the lounge trvs now get hot water and control to the set point which is 0.5C above what it was! To me I can notice that and not satisfactory.  What I did was at bedroom schedule time lowered the lounge set points by 0.5C, not 100% if another room calls early but works most of the time!  I have ha automation instead of hive HoD so can modify the point at which the boiler is called etc. work in progress but delayed by new boiler.  This is another reason to get temperature compensation working so trvs have less to do.

**Reply by chrisbev** [reply]
From earlier.  I have flow and return pipe temperatures into a raspberry pico so have plots and then wifi to ha, isn't ha good for yhat?  I had wondered about the dhw return temperature being quite high and just thought an old cylinder with scale?  Fitted a water softener as very hard water here and no noticeably difference so your restricter could be worth a consideration on dhw, the boiler is quite low modulation so I thought higher flow the better but that presumably is false thinking?
It is a 625 (not a 620) and the setting says 25.5 the reason for this was vaillant's way of overcoming some shortfall in the boiler algorithm (I believe it is a new algorithm for this updated model but same name ecoTEC Plus?).  
For problem 1 the booler modulation was ~20% with flow temp 40C and didn’t drive to set point setting D.000 to 25.5 rather than auto fixed that but it still modulates to get up to/down to set point just doesn’t stick down there (I assumed remembering the last load conditions) maybe after hours/days it might 'recover'?  I think this is where my 625 differs from previous 428 that had the reverse problem it fired full blast and with low load ran into high temp trip before it modulated low.  Setting D.000 lower had no effect I was told it had to fire high for a certain time to ensure good ignition, the new one has IoniDetect for mixture control for different modulations so can keep good combustion to a lower level to get the 10:1 range and presumably start low and then increase (or supposed to).  All be it they haven’t got it quite right for my system hence the fudge?
D.001/002 I have not changed and at defaults 5 & 20. 
D.071 for me is max target flow temperature not flow rate and at defaults 75C.
D.128 min flow temp not rate and at 30C (default 40C but not changed by me).
With this ufh blender the hot port is to boiler so if satisfied (never quite gets there?) would shut to the boiler so has to have a separate pump?  Common port goes off to the ufh pump and floor coils.  Cold port tee'd into coils back to boiler pipe. To use the boiler pump I expect boiler temp is controlled and no blender?
Problem 2, fixing the boiler pump stage to me is also a hack to get around the boiler algorithm not being suitable for my sysem or could say it the other way round but I did ask the installer (and later vaillant) if the boiler was suitable for the zones dhw, rads and ufh?
With the 2 hacks now seems to start and get going on the first try.  Am I a happy Customer?  Yes and No I suppose?

**Reply by JonesGW-MGD** [reply]
sorry my bad - not sure why I read 620 instead of 625 and yes d.071 and d.128 - I meant temps not rate my bad again.

If setting d.000 works for you - then hey - stick with it!   These devices are 'black boxes' and we'll never know what they really do so it's very much trial and error - my setup is working well but one of the reasons I ventured into ebusd/HA is I think (ironically since I wsa determined to have a really low flow, return and dt!) my delta T is too low (might benefit from an external bypass?) and my D.029 flow rate (confusingly displayed on screen in litres and different metric in HA) seems too high so always learning!

I think your (Vaillant's!) pump hack is more suspect - of course running the pump high and fast will work - like they did decades ago - just not sure it's the best your setup deserves!  You could start experimenting here though with data feedback from HA.

I think your setup is obviously capable - just not sure it (/Vaillant) 'plays well with others'. (- annoyingly they don't comply with OpenTherm - except in the NL where they legally have to - grrrr!! Try buying a Vaillant VR33 OpenTherm module in the UK....) - but to fix that you'd need to replace your controls with a VR71 ebus wiring centre, a VRC720f or similar controller and if you wanted see the results remotely - a VR920/1 internet gateway- basically another 700 quid plus install!  Oh, and another 100 quid per ambiSENSE TRV!....You seem to be fairly close to where you need to be - at least with HA and ebusd you can collect a sh*t-tonne of data which is what I like!

**Comment by chrisbev** [comment]
Thanks again just a quick question on your graph if I may then just as I have the ha data of my boiler in ha all-be-it some entries repeated a few times and some entries with the same name have different values to sort and about to try receiving and sending single lines ready to add to automations I have to take a break from ha for a week or so but considering the green flow, yellow return and blue required/set point lines.  I see green and yellow close together for a lot of time so assume boiler is off or not heating, fair enough but why is the blue line above and not seem to tie up for most of the time?  Also there is one point 04/10 01:00 ish where return is above flow and the required dips?  Middle of night? There is at 5:00 ehat looks like it could be the required rising and flow climbing but why so slow for flow to get to required temperature at 8:00?  Just asking to try and understand the logic not a criticism.

**Reply by JonesGW-MGD** [reply]
To. answer your question, I'm not fully sure of the behaviour of my system!  From an 'input' perspective, each TRV has it's own programme but broadly - heating is off overnight (23:00-07:00 - setback to 15 or so), downstairs (and some upstairs) is warmed 08:00-10:00 (20-21) and then sets back to 15-16 or so and then back on to 20-ish for the evening.

This might be a better graph - longer period, colder period (end of Feb, early March)

<img width="2116" alt="image" src="https://github.com/john30/ebusd/assets/68991992/e37bf0f0-5446-4bf8-a379-fcfad0603473">

I don't understand why the 'DesiredTenp'  'oscillates' at night (could be the frequency of plot is off-putting) - obviously as the outside temp falls, the target flow is going to shift anticipating satisfying setback temp?.
I'm not too surprised the flow lags the desired - I read that as a function of the low-ish heat curve but sometimes it leads according the chart?
I've always been concerned the flow and return DT is really low.  Low is generally good particularly since my objective is to run a low temperature heating system but the usual recommendation is DT = 0.3 x Flow, it implies the heat isn't given a chance to 'emit' but the rads definitely warm the rooms.

I have zigbee temp gauges in most rooms but haven't yet overlaid that data.

The Vaillant setup claims to be 'adaptive' and the TRVs are certainly very 'chatty' with the controller, making audible regular adjustments - but it's hard to work out what is 'learning' algorithms and what is unwanted behaviour

TBH generally happy with the system - was such a 'low bar' coming from electric.  Have also had to resolve some issues left by installer - like  air noise since they didn't 'believe in airvents'...  Just need to find time to leverage the ebusd data to optimise further!
